src/main/java/com/addthis/codec/jackson/CodecTypeDeserializer.java

```diff
@@ -125,7 +125,9 @@ public Object deserializeTypedFromAny(JsonParser jp, DeserializationContext ctxt
             if (bean != null) {
                 return bean;
             } else {
-                JsonDeserializer<Object> deser = _findDeserializer(ctxt, null);
+                // Jackson 2.6+ throws NPE on null typeId parameter (underlying Map changed from HashMap
+                // to ConcurrentHashMap), so use empty string instead of null
+                JsonDeserializer<Object> deser = _findDeserializer(ctxt, "");
```

Comment:
Pretty sure that statement always resolved into getting the deserializer for the default impl, and we can just do that directly.